solution/index.js

```diff
@@ -0,0 +1,316 @@
+        {
+        list.style.display = "none";
+        }else
+        {
+            list.style.display = "";
+        }
+    }
+}
+
+let dragged;
+
+document.addEventListener("dragstart", function( event ) {
+    dragged = event.target; // store a ref. on the dragged elem
+    event.target.style.opacity = .5; // make it half transparent
+}, false);
+
+document.addEventListener("dragend", function( event ) {
+    event.target.style.opacity = ""; // reset the transparency
+}, false);
+
+ /* events fired on the drop targets */
+ document.addEventListener("dragover", function( event ) {
+    // prevent default to allow drop
+    event.preventDefault();
+}, false);
+
+document.addEventListener("dragenter", function( event ) {
+    // highlight potential drop target when the draggable element enters it
+    if ( event.target.className === "dropzone" ) {
+        event.target.style.background = "purple";
+    }
+}, false);
+
+document.addEventListener("dragleave", function( event ) {
+    // reset background of potential drop target when the draggable element leaves it
+    if ( event.target.className === "dropzone" ) {
+        event.target.style.background = "";
+    }
+}, false);
+
+document.addEventListener("drop", function( event ) {
+    // prevent default action (open as link for some elements)
+    event.preventDefault();
+    // move dragged elem to the selected drop target
+    if ( event.target.className === "dropzone" ) 
+    {
+        event.target.style.background = "";
+        // before the change in the dom 
+        if(dragged.parentNode.className === "to-do-tasks")
+        {
+            let i = toDoStorage.indexOf(dragged.innerText);
+            toDoStorage.splice(i,1);
+        }
+        else if(dragged.parentNode.className === "in-progress-tasks")
+        {
+            let i = inProgressStorage.indexOf(dragged.innerText);
+            inProgressStorage.splice(i,1);
+        }
+        else if(dragged.parentNode.className === "done-tasks")
+        {
+            let i = doneStorage.indexOf(dragged.innerText);
+            doneStorage.splice(i,1);
+        }
+        // change in the dom
+        dragged.parentNode.removeChild( dragged );
+        event.target.children[0].insertBefore(dragged , event.target.children[0].firstChild);
+        // after the change in the dom 
+        if(dragged.parentNode.className === "to-do-tasks")
+        {
+            toDoStorage.unshift(dragged.innerText);
+        }
+        else if(dragged.parentNode.className === "in-progress-tasks")
+        {
+            inProgressStorage.unshift(dragged.innerText);
+        }
+        else if(dragged.parentNode.className === "done-tasks")
+        {
+            doneStorage.unshift(dragged.innerText);
+        }
+        localStorage.setItem("tasks" ,JSON.stringify(localStorageObj));
+    }
+}, false);
+
+document.getElementById("save-btn").addEventListener("click",saveToApi);
+async function saveToApi(){
```

Comment:
some of the functions should get variables passed to it, instead of global variables.